src/Compilers/CSharp/Test/Semantic/Semantics/RecordStructTests.cs

```diff
@@ -1526,15 +1527,71 @@ unsafe record struct C(
                 Diagnostic(ErrorCode.ERR_FieldCantBeRefAny, "System.TypedReference").WithArguments("System.TypedReference").WithLocation(14, 5),
                 // 0.cs(15,5): error CS8345: Field or auto-implemented property cannot be of type 'RefLike' unless it is an instance member of a ref struct.
                 //     RefLike P9, // 8
-                Diagnostic(ErrorCode.ERR_FieldAutoPropCantBeByRefLike, "RefLike").WithArguments("RefLike").WithLocation(15, 5),
-                // 0.cs(16,21): error CS8908: The type 'delegate*<void>' may not be used for a field of a record.
-                //     delegate*<void> P10); // 9
-                Diagnostic(ErrorCode.ERR_BadFieldTypeInRecord, "P10").WithArguments("delegate*<void>").WithLocation(16, 21)
-                );
+                Diagnostic(ErrorCode.ERR_FieldAutoPropCantBeByRefLike, "RefLike").WithArguments("RefLike").WithLocation(15, 5)];
+            comp.VerifyDiagnostics(expected);
+            comp.VerifyEmitDiagnostics(expected);
         }
+        [Fact]
+        [WorkItem(48115, "https://github.com/dotnet/roslyn/issues/48115")]
+        [WorkItem("https://github.com/dotnet/roslyn/issues/66312")]
+        public void RestrictedTypesAndPointerTypes_WithEquals()
```

Comment:
>RestrictedTypesAndPointerTypes_WithEquals [](http://example.com/codeflow?start=20&length=41)

Consider combining with previous test with `[Theory]` and `bool withEquals` parameter, so it's clear we expect the behavior to be the same.